tests/utils/utils.go

```diff
@@ -0,0 +1,333 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package utils
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"fmt"
+	"math/big"
+	"strconv"
+	"strings"
+
+	"github.com/ava-labs/avalanchego/ids"
+	avalancheWarp "github.com/ava-labs/avalanchego/vms/platformvm/warp"
+	warpBackend "github.com/ava-labs/subnet-evm/warp"
+
+	"github.com/ava-labs/awm-relayer/messages/teleporter"
+	"github.com/ava-labs/subnet-evm/core/types"
+	"github.com/ava-labs/subnet-evm/ethclient"
+	"github.com/ava-labs/subnet-evm/interfaces"
+	"github.com/ava-labs/subnet-evm/params"
+	"github.com/ava-labs/subnet-evm/tests/utils"
+	"github.com/ava-labs/subnet-evm/tests/utils/runner"
+	predicateutils "github.com/ava-labs/subnet-evm/utils/predicate"
+	"github.com/ava-labs/subnet-evm/x/warp"
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/ethereum/go-ethereum/log"
+	. "github.com/onsi/gomega"
+)
+
+var (
+	DefaultTeleporterTransactionGas       uint64 = 200_000
+	DefaultTeleporterTransactionGasFeeCap        = big.NewInt(225 * params.GWei)
+	DefaultTeleporterTransactionGasTipCap        = big.NewInt(params.GWei)
+	DefaultTeleporterTransactionValue            = common.Big0
+)
+
+type SendCrossChainMessageEvent struct {
+	DestinationChainID ids.ID
+	MessageID          *big.Int
+	Message            teleporter.TeleporterMessage
+}
+
+type ReceiveCrossChainMessageEvent struct {
+	OriginChainID ids.ID
+	MessageID     *big.Int
+	Message       teleporter.TeleporterMessage
+}
+
+// Teleporter contract sendCrossChainMessage input type
+type SendCrossChainMessageInput struct {
+	DestinationChainID      ids.ID
+	DestinationAddress      common.Address
+	FeeInfo                 FeeInfo
+	RequiredGasLimit        *big.Int
+	Message                 []byte
+	AllowedRelayerAddresses []common.Address
+}
+
+type FeeInfo struct {
+	ContractAddress common.Address
+	Amount          *big.Int
+}
+
+//
+// Test utility functions
+//
+
+// Subscribes to new heads, sends a tx, and waits for the new head to appear
+// Returns the new head
+func SendTransactionAndWaitForAcceptance(
+	ctx context.Context,
+	wsClient ethclient.Client,
+	tx *types.Transaction) (*types.Header, *types.Receipt) {
+
+	newHeads := make(chan *types.Header, 1)
+	subA, err := wsClient.SubscribeNewHead(ctx, newHeads)
+	Expect(err).Should(BeNil())
+	defer subA.Unsubscribe()
+
+	err = wsClient.SendTransaction(ctx, tx)
+	Expect(err).Should(BeNil())
+	newHead := <-newHeads
+
+	receipt, err := wsClient.TransactionReceipt(ctx, tx.Hash())
+	Expect(err).Should(BeNil())
+	Expect(receipt.Status).Should(Equal(types.ReceiptStatusSuccessful))
+
+	return newHead, receipt
```

Comment:
We probably don't have to return `newHead` here, the receipt has the blockhash included.